libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFDefaultUserAccountPersister.m

```diff
@@ -0,0 +1,306 @@
+                                         code:SFUserAccountManagerCannotWriteUserData
+                                     userInfo:@{ NSLocalizedDescriptionKey : reason } ];
+        return NO;
+    }
+    // Save it.
+    BOOL saveFileSuccess = [manager createFileAtPath:filePath contents:encryptedArchiveData attributes:@{ NSFileProtectionKey : [SFFileProtectionHelper fileProtectionForPath:filePath] }];
+    if (!saveFileSuccess) {
+        NSString *reason = [NSString stringWithFormat:@"Could not create user account data file at path.  %@",filePath];
+        [self log:SFLogLevelWarning msg:reason];
+        if (error)
+            *error = [NSError errorWithDomain:SFUserAccountManagerErrorDomain
+                                         code:SFUserAccountManagerCannotWriteUserData
+                                     userInfo:@{ NSLocalizedDescriptionKey : reason } ];
+        return NO;
+    }
+
+    return YES;
+}
+
+
+
+/** Loads a user account from a specified file
+ @param filePath The file to load the user account from
+ @param account On output, contains the user account or nil if an error occurred
+ @param error On output, contains the error if the method returned NO
+ @return YES if the method succeeded, NO otherwise
+ */
+ - (BOOL)loadUserAccountFromFile:(NSString *)filePath account:(SFUserAccount**)account error:(NSError**)error {
+
+        NSFileManager *manager = [[NSFileManager alloc] init];
+        NSString *reason = @"User account data could not be decrypted. Can't load account.";
+        NSData *encryptedUserAccountData = [manager contentsAtPath:filePath];
+        if (!encryptedUserAccountData) {
+            reason = [NSString stringWithFormat:@"Could not retrieve user account data from '%@'", filePath];
+            if (error) {
+                *error = [NSError errorWithDomain:SFUserAccountManagerErrorDomain
+                                             code:SFUserAccountManagerCannotRetrieveUserData
+                                         userInfo:@{NSLocalizedDescriptionKey: reason}];
+            }
+            [self log:SFLogLevelDebug msg:reason];
+            return NO;
+        }
+        SFEncryptionKey *encKey = [[SFKeyStoreManager sharedInstance] retrieveKeyWithLabel:kUserAccountEncryptionKeyLabel keyType:SFKeyStoreKeyTypeGenerated autoCreate:YES];
+        NSData *decryptedArchiveData = [SFSDKCryptoUtils aes256DecryptData:encryptedUserAccountData withKey:encKey.key iv:encKey.initializationVector];
+        if (!decryptedArchiveData) {
+            if (error) {
+                *error = [NSError errorWithDomain:SFUserAccountManagerErrorDomain
+                                             code:SFUserAccountManagerCannotRetrieveUserData
+                                         userInfo:@{NSLocalizedDescriptionKey: reason}];
+            }
+            [self log:SFLogLevelWarning msg:reason];
+            return NO;
+        }
+
+        @try {
```

Comment:
Now that iOS 9 is our minimum, let's just simplify by ditching the try/catch block, in favor of the logic in the try section.